### PR TITLE
[FAB-15198] Document peer crash scenario when indexes are not installed

### PR DESCRIPTION
**Pull request by nikhil550**
Signed-off-by: NIKHIL E GUPTA <negupta@us.ibm.com>

#### Type of change

- Documentation update

#### Description

In the new chaincode lifecycle, it is possible that the indexes will not deploy if the peer crashes after the chaincode is installed. It was decided that this would be fixed in documentation for the moment.

Other fixes to this CouchDB as state database doc:
- Edit introduction
- Update for Fabric Chaincode lifecycle
- Make language stronger around the use of indexes
- Include port mapping note currently written in the first network tutorial

### CODE REVIEW

File: docs/source/couchdb_as_state_database.rst [hunk 1]
```diff
@@ -223,6 +229,13 @@ Docker compose scripts only set the username and password at the creation of
 the container. The *local.ini* file must be edited if the username or password
 is to be changed after creation of the container.
 
+If you choose map the fabric-couchdb container port to a host port, make sure you
```
Comment (denyeart) [inline docs/source/couchdb_as_state_database.rst]: "choose to map"

File: docs/source/couchdb_as_state_database.rst [hunk 2]
```diff
@@ -223,6 +229,13 @@ Docker compose scripts only set the username and password at the creation of
...
+are aware of the security implications. Mapping the CouchDB container port in a
+development environment exposes the CouchDB REST API and allows you to visualize
+the database via the CouchDB web interface (Fauxton). In a production environment
+you should refrain from mapping the host port to restrict access to the CouchDB
```
Comment (denyeart) [inline docs/source/couchdb_as_state_database.rst:235]: "restrict external client access to the CouchDB container"

...the peer container will still have access of course.

Comment (nikhil550) [reply docs/source/couchdb_as_state_database.rst:235]: Added: Only the peer will be able to access the CouchDB container.

**Review by nikhil550 — COMMENTED** [review]
@denyeart Rephrased rich query and commit time statement: "However,
there is no guarantee that the result set for a rich query will be stable between
the chaincode execution and commit time. As a result, you should not use a rich query and
update the channel ledger in a single transaction. "

Rm comprehensive: "This makes it easier for your applications and
queries are excellent for understanding what is on the ledger. Proposal responses for these types
chaincode read the data stored on the blockchain ledger."

Also removed allows: "If you use CouchDB and model your data in JSON, you can also deploy indexes with your chaincode ...."

File: docs/source/couchdb_as_state_database.rst [hunk 3]
```diff
@@ -18,23 +18,18 @@ key of ``owner,asset_id`` can be used to query all assets owned by a certain ent
...
-between chaincode execution and commit time for rich queries, and therefore rich queries
-are not appropriate for use in update transactions, unless your application can guarantee the
-result set is stable between chaincode execution time and commit time, or can handle potential
-changes in subsequent transactions. For example, if you perform a rich query for all assets
-owned by Alice and transfer them to Bob, a new asset may be assigned to Alice by another
-transaction between chaincode execution time and commit time, and you would miss this "phantom"
-item.
+Modeling your data in JSON allows you to issue rich queries against the values of your data,
+instead of only being able to query the keys. This makes it easier for your applications and
+chaincode read the data stored on the blockchain ledger. Using CouchDB can help you meet
```
Comment (denyeart) [inline docs/source/couchdb_as_state_database.rst]:
```suggestion
chaincode to read the data stored on the blockchain ledger. Using CouchDB can help you to address
```

**Comment by denyeart** [comment]
Looks good after that final typo is fixed.

**Comment by nikhil550** [comment]
> Looks good after that final typo is fixed.

done.

**Review by denyeart — APPROVED** [review]

**Pull request merged**